internal/rset/tx.go

```diff
@@ -0,0 +1,66 @@
+package rset
+
+import (
+	"database/sql"
+	"github.com/nalgeon/redka/internal/core"
+	"github.com/nalgeon/redka/internal/sqlx"
+	"time"
+)
+
+var sqlAdd = []string{`
+insert into rkey (key, type, version, etime, mtime)
+values (:key, :type, :version, :etime, :mtime)
+on conflict (key) do update set 
+    version = version + 1,
+    type = excluded.type,
+    etime = excluded.etime,
+    mtime = excluded.mtime
+;`,
+
+	`insert into rset (key_id, elem)
+values ((select id from rkey where key = :key), :value)`,
+}
+
+// Tx is a set repository transaction.
+type Tx struct {
+	tx sqlx.Tx
+}
+
+func NewTx(tx sqlx.Tx) *Tx {
+	return &Tx{tx}
+}
+
+// Add adds key elems to set.
+func (t *Tx) Add(key string, elems ...any) (int, error) {
+	return t.add(key, elems...)
+
+}
+
+func (t *Tx) add(key string, elems ...any) (int, error) {
```

Comment:
What is the need for this method?

Reply:
I thought that we may need to reuse the same method for other commands, but I might be overengineering. I can remove it.